doc/ui-guidelines.md

```diff
@@ -0,0 +1,257 @@
+            on-blur                (fn []
+                                     (reset! focused? false))]))))
+```
+
+NOW:
+```clojure
+(defn view
+  []
+  (let [[focused? set-focused] (rn/use-state false)
+        on-clear               (rn/use-callback
+                                (fn []
+                                  (set-status (if focused? :active :default)))
+                                [focused?])
+        on-focus               (rn/use-callback
+                                (fn []
+                                  (set-focused true)))
+        on-blur                (rn/use-callback
+                                (fn []
+                                  (set-focused false)))]))
+```
+
+> [!IMPORTANT]
+> Instead of a regular atom, we also use a state hook. It's important to understand that its value is now encapsulated within the function, and we must pass it as a dependency to `use-callback` so that the function re-executes with the new value.
+
+## Effects
+
+In most cases, the effect should be used for implementing visual reactions (such as animations) to state changes or for lifecycle methods.
+
+ANIMATION:
+```clojure
+(defn f-zoom-button
+  [{:keys [selected? current-zoom]}]
+  (let [size (reanimated/use-shared-value (if selected? 37 25))]
+    (rn/use-effect #(reanimated/animate size (if selected? 37 25)) [current-zoom])
+    [rn/touchable-opacity
+     {:style (style/zoom-button-container size)}]))
+```
+
+This effect will be run only when current-zoom has changed.
+
+LIFECYCLE:
+
+```clojure
+(defn view
+  [{:keys []}]
+  (let [opacity (reanimated/use-shared-value 0)]
+    (rn/use-mount #(reanimated/animate opacity 1))
+    [rn/view
+     {:style (style/opacity opacity)}]))
+```
+
+```clojure
+(defn view
+  [{:keys []}]
+  (let []
+    (rn/use-unmount #(rn/dispatch [:unmounted]))
+    [rn/view]))
+```
+
+> [!IMPORTANT]
+> Effects shouldn't be used as a reaction to state changes for altering logic.
+
+BAD:
+
+```clojure
+(defn view
+  [collectible-list (rf/sub [:wallet/all-collectibles])]
+  (let []
+    (rn/use-effect #(rn/dispatch [:all-collectibles-changed]) [collectible-list])
+    [rn/view]))
+```
+
+Instead `:all-collectibles-changed` shdoul be used in the handler which changes `collectible-list` state
+
+## Performance tips
+
+To begin with, we need to understand that there are two distinct stages for a component: creation and update. React creates a render tree, a UI tree, composed of the rendered components.
+
+![react_tree.png](react_tree.png)
+
+### Component creation
+
+For component creation, the most critical factor is the number of elements involved, so we should strive to minimize them. For instance, it's advisable to avoid using unnecessary wrappers or containers.
+
+BAD:
+
+```clojure
+(defn view
+  []
+  (let []
+    [rn/view {:padding-top 20}
+     [quo/button]]))
+```
+
+GOOD:
+```clojure
+(defn view
+  []
+  (let []
+    [quo/button {:style {:padding-top 20}}]))
```

Comment:
for all of our quo components we have been using a `container-style` prop which goes for this very purpose. 
Can you update this to: `[quo/button {:container-style {:padding-top 20}}` ?

Reply:
thanks